### PR TITLE
[Python] Implement TLV List type

### PR DESCRIPTION
**Pull request by erjiaqing**
The Python library has a limited support of the TLV list type, which is a TLV container supports anonymous tags.

This PR enhances the TLV module in the Python library to have a better support of TLV Lists in CHIP TLV.

## API Implemented (also in the doc string)

A TLVList can be constructed from a `list` of tuples of tag and value. `None` stands for "anonymous tag".
e.g.
```
l = TLVList([(1, 'a'), (2, 'b'), (None, 'c')])
```
Constructs a list of three items, tag 1 is 'a', tag 2 is 'b' and with an anonymous item 'c'.
Since TLVLists are ordered, it is meanful to iterate over an list:
e.g.
```
for tag, val in l:
    print(f"tag={tag}, val={val}")
```
Outputs:
```
tag=1, val=a
tag=2, val=b
tag=None, val=c
```
One can also append items into an list:
e.g.
```
l.append(3, 'd')
```
The content of `l` will be `[(1, 'a'), (2, 'b'), (None, 'c'), (3, 'd')]`
One can access an item in the list via the tag.
e.g.
```
val = l[1]
# val is 'a'
```
It is also possible to get an item via the index since it is ordered:
e.g.
```
tag, val = l[TLVList.IndexMethod.Tag:2]
# tag is None, val is 'c'
```

## Tests

See src/controller/python/test/unit_tests/test_tlv.py

### CODE REVIEW

File: src/controller/python/BUILD.gn [hunk 1]
```diff
@@ -350,6 +350,7 @@ chip_python_wheel_action("chip-clusters") {
         "chip/clusters/TestObjects.py",
         "chip/clusters/Types.py",
         "chip/tlv/__init__.py",
+        "chip/tlv/tlvlist.py",
```
Comment (andy31415) [inline src/controller/python/BUILD.gn:353]: tlv/tlvlist.py seems to stutter. could we just name it list ?

**Review by andy31415 — APPROVED** [review]

**Review by yunhanw-google — APPROVED** [review]

File: src/controller/python/BUILD.gn (hunk 1 above)
Comment (erjiaqing) [reply src/controller/python/BUILD.gn]: makes sense

**Comment by andy31415** [comment]
@erjiaqing my suggesion of using "list" name may have conflicted with the python 'list' type :( I see a lot of CI errors ... if that is the cause, may revert. It stutters, but at least it works.

**Comment by tcarmelveilleux** [comment]
Not stale

**Review by woody-apple — APPROVED** [review]

**Pull request merged**